test/store_unittest.cc

```diff
@@ -0,0 +1,234 @@
+#include "../src/configuration.h"
+#include "../src/resource.h"
+#include "../src/store.h"
+#include "../src/time.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataStoreTest : public ::testing::Test {
+ protected:
+  Configuration config;
+  MetadataStore store;
+
+  MetadataStoreTest() : config(), store(config) {}
+
+  std::map<MonitoredResource, MetadataStore::Metadata> GetMetadataMap() const {
+    return store.GetMetadataMap();
+  }
+
+  void PurgeDeletedEntries() {
+    store.PurgeDeletedEntries();
+  }
+
+};
+
+TEST_F(MetadataStoreTest, ResourceWithOneIdStoredCorrectly) {
```

Comment:
`CorrectlyStored` or `StoredCorrectly`? Either is fine, but please pick one and stick with it. You seem to be preferring `CorrectlyStored` throughout.